openwisp_notifications/handlers.py

```diff
@@ -203,25 +203,61 @@ def send_email_notification(sender, instance, created, **kwargs):
     else:
         target_url = None
     if target_url:
-        description += _('\n\nFor more information see %(target_url)s.') % {
-            'target_url': target_url
-        }
-
-    send_email(
-        subject,
-        description,
-        instance.message,
-        recipients=[instance.recipient.email],
-        extra_context={
-            'call_to_action_url': target_url,
-            'call_to_action_text': _('Find out more'),
-        },
-    )
+        description += _('\n\nFor more information see %(target_url)s.') % {'target_url': target_url}
 
-    # flag as emailed
-    instance.emailed = True
-    # bulk_update is used to prevent emitting post_save signal
-    Notification.objects.bulk_update([instance], ['emailed'])
+    return True, subject, description, target_url
+
+
+@receiver(post_save, sender=Notification, dispatch_uid='send_email_notification')
+def send_email_notification(sender, instance, created, **kwargs):
+    # Abort if a new notification is not created
+    if not created:
+        return
+
+    can_send, subject, description, target_url = can_send_email(instance)
+
+    if not can_send:
+        return
+
+    recipient_id = instance.recipient.id 
+    cache_key = f'email_batch_{recipient_id}'
+    cache_data = cache.get(cache_key, {'last_email_sent_time': None, 'batch_scheduled': False})
+    is_send_email = False
+
+    if cache_data['last_email_sent_time']:
```

Comment:
I think we could simplify these conditionals. 

If `cache.get(cache_key)` returns `None`, this means one of  the two things 
1. No email was sent to the user 
2.  The `EMAIL_BATCH_INTERVAL` has passed because you are setting a timeout to the cache in this line 
```
cache.set(cache_key, cache_data, timeout=EMAIL_BATCH_INTERVAL)
```

I think, we can handle both of these scenarios in the same condition.

What do you think @Dhanus3133? Will it cover all the scenarios?